### ISSUE TITLE
ld: symbol(s) not found for architecture x86_64 Mac issue

### ISSUE THREAD

**Issue by prateektiwari7** [bitshares/bitshares-core#1545]
Hi I am facing issue in building the code.

`[ 29%] Linking CXX executable real128_test
Undefined symbols for architecture x86_64:
  "boost::system::system_category()", referenced from:
      fc::copy(fc::path const&, fc::path const&) in libfc.a(filesystem.cpp.o)
      boost::filesystem::detail::recur_dir_itr_imp::increment(boost::system::error_code*) in libfc.a(filesystem.cpp.o)
      boost::filesystem::detail::recur_dir_itr_imp::push_directory(boost::system::error_code&) in libfc.a(filesystem.cpp.o)
      fc::udp_socket::~udp_socket() in libfc.a(udp_socket.cpp.o)
      fc::udp_socket::send_to(char const*, unsigned long, fc::ip::endpoint const&) in libfc.a(udp_socket.cpp.o)
      fc::udp_socket::send_to(std::__1::shared_ptr<char const> const&, unsigned long, fc::ip::endpoint const&) in libfc.a(udp_socket.cpp.o)
      fc::udp_socket::open() in libfc.a(udp_socket.cpp.o) <br>
 
  "boost::system::generic_category()", referenced from:
      boost::thread::join() in libfc.a(thread.cpp.o)
      boost::condition_variable::condition_variable() in libfc.a(thread.cpp.o)
      boost::mutex::mutex() in libfc.a(thread.cpp.o)
      boost::condition_variable::wait(boost::unique_lock<boost::mutex>&) in libfc.a(thread.cpp.o)
      boost::mutex::lock() in libfc.a(thread.cpp.o)
      boost::unique_lock<boost::mutex>::unlock() in libfc.a(thread.cpp.o)
      boost::unique_lock<boost::mutex>::lock() in libfc.a(thread.cpp.o) 

**ld: symbol(s) not found for architecture x86_64**
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libraries/fc/tests/real128_test] Error 1
make[1]: *** [libraries/fc/tests/CMakeFiles/real128_test.dir/all] Error 2
make: *** [all] Error 2`

I have MAC OSX 10.12 

I already went through the build instruction steps of **https://github.com/bitshares/bitshares-core/wiki/Building-on-OS-X**

Please help me to find build.

**Comment by jmjatlanta** [comment]
Can you tell me which version of Boost you have installed? Also, please let me know the parameters that you passed to the ``cmake`` command.

**Comment by prateektiwari7** [comment]
This is `cmake` command I am using `cmake -DBOOST_ROOT=/usr/local/opt/boost@1.60 -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl .`

For Boost Please check this screenshot
![screen shot 2019-01-29 at 10 47 33 am](https://user-images.githubusercontent.com/25363351/51890689-b2afbd00-23c2-11e9-815c-5e2c8e66d131.png)

**Comment by prateektiwari7** [comment]
This is `cmake` command I am using `cmake -DBOOST_ROOT=/usr/local/opt/boost@1.60 -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl .`

For boost please check this screenshot.

![screen shot 2019-01-29 at 12 39 21 pm](https://user-images.githubusercontent.com/25363351/51890786-015d5700-23c3-11e9-80a9-f325a6f81fa3.png)

**Comment by prateektiwari7** [comment]
I got another Error now. 

My steps to build are .

```
1.brew install boost cmake git openssl autoconf automake libtool 
2.brew link --force openssl
3.git clone https://github.com/bitshares/bitshares-core.git
4.cd bitshares-core
5.git submodule update --init --recursive
6.cmake .
7.make
``` 

Currently I reinstall my boost version to 
```
Prateek-Tiwari-2:bitshares-core yudiz$ brew reinstall boost
==> Reinstalling boost 
==> Downloading https://homebrew.bintray.com/bottles/boost-1.68.0_1.sierra.bottle.tar.gz
Already downloaded: /Users/yudiz/Library/Caches/Homebrew/downloads/7c649ada5214d3f047263ab765fb2992cdda477e083e1545a7c3218dfc8c54d0--boost-1.68.0_1.sierra.bottle.tar.gz
==> **Pouring boost-1.68.0_1.sierra.bottle.tar.gz**
 /usr/local/Cellar/boost/1.68.0_1: 13,712 files, 460.0MB
```

The error now I am getting as 
`In file included from /Users/yudiz/Documents/Graphene/bitshare_!/bitshares-core/libraries/fc/src/thread/thread_d.hpp:5:
/Users/yudiz/Documents/Graphene/bitshare_!/bitshares-core/libraries/fc/src/thread/context.hpp:3:10: fatal error: 
      'boost/context/all.hpp' file not found
#include <boost/context/all.hpp>
         ^
4 warnings and 1 error generated.`

**Comment by abitmore** [comment]
IIRC bitshares-core doesn't support boost 1.68 so far.

**Comment by jmjatlanta** [comment]
Yes, Boost 1.68 moved around some of the headers. The fixes are in development, but not quite ready yet.

It looks like you have Boost 1.60 installed, and that is what I currently use on macOS. I will follow your steps above, and see if I can recreate your issue.

**Comment by prateektiwari7** [comment]
Hi @jmjatlanta Did you fixed that ?

**Comment by pmconrad** [comment]
Boost 1.68 is not supported yet. Please use a compatible version for now.

**Comment by jmjatlanta** [comment]
Please provide all the output from the following procedure:
```
rm CMakeCache.txt
rm -Rf CMakeFiles
cmake -DBOOST_ROOT=/usr/local/opt/boost@1.60 -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl .
make
```
I am on a slightly newer version of macOS, but I do not think anything recent would have caused problems for your version.

**Comment by jmjatlanta** [comment]
The real fix for smart_ref can be tracked by following issue #1548.

**Comment by jmjatlanta** [comment]
Closed by mistake... Sorry. Please let us know if this is still an issue.

**Comment by prateektiwari7** [comment]
@jmjatlanta Thanks for your help

**Comment by jmjatlanta** [comment]
Fixed by PR #1556 which is part of Issue #1548
